src/main/java/mate/academy/model/User.java

```diff
@@ -53,8 +57,25 @@ public void setSalt(byte[] salt) {
     @Override
     public String toString() {
         return "User{"
-            + "id=" + id
-            + ", email='" + email + '\''
-            + '}';
+                + "id=" + id
+                + ", email='"
+                + email + '\''
+                + '}';
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        User user = (User) o;
+        return Objects.equals(id, user.id)
+                && Objects.equals(email, user.email)
+                && Objects.equals(password, user.password);
```

Comment:
Consider excluding `password` and `salt` from the `equals` method to avoid potential security risks and ensure equality checks are based on non-sensitive fields.